lua/spectre_common.lua

```diff
@@ -93,8 +110,13 @@ local function determine_if_cacheable(url, namespace, request_headers)
         return cacheability_info
     end
 
+    local http_method = ngx.req.get_method()
     for cache_name, cache_entry in pairs(spectre_config['cached_endpoints']) do
-        if ngx.re.match(url, cache_entry['pattern']) then
+        if ngx.re.match(url, cache_entry['pattern']) and (
+            -- Compute if http_method is same as in config or http method is GET
+            cache_entry['request_method'] == http_method or cacheability_info.cache_entry.request_method == http_method
```

Comment:
what's the difference between `cache_entry['request_method']` and `cacheability_info.cache_entry.request_method`? Aren't they the same thing?

Reply:
cacheability_info.cache_entry.request_method is the default method. only after this conditional is passed, we assign cacheability_info.cache_entry = cache_entry. Will rename variables to make it more explicit.